### PR TITLE
Fix GDScript checking for assigning to a constant only in release

### LINKED ISSUE #25334: Godot fails loading compiled GDScript in release build when assigning new value to iterator
**Godot version:**
6e600706e9c0d2d08ea4cf57b7cda423de1a45fd

**OS/device including version:**
Windows 10, 64-bit

godot displayed no errors at all :/. just opened and closed. (even running in cmd and piping output to a file). so i removed every single piece of code one by one, and exported it each time, etc.. for a few hours lol. finally found it!

**Issue description:**
 - `player_id = int(player_id)`  will cause Godot to open and close without any error:
 - **Exported Godot apps only**. Only happens when Script Mode is set to **Compiled** or **Encrypted**.
 - This **does not happen** if Export To Debug is enabled, or running from the editor

```
extends Control

func _ready():
	var message = {"2": "fakedatahere"}
	for player_id in message:
		player_id = int(player_id)
```
**Minimal reproduction project:**
[New Game Project66.zip](https://github.com/godotengine/godot/files/2798725/New.Game.Project66.zip)

### LINKED ISSUE #26412: Release builds can crash from certain loop operations.
**Godot version:**

996af80

**OS/device including version:**

Win10 64-bit

**Issue description:**

References to Autoloads will go null in release builds if you try to perform operations on the iterator element of a for loop.  Wasn't seeing it until now, because it is not an issue in debug builds.

If your project has references to an Autoload's methods, the release build will crash without indications.

Haven't fully explored it, so maybe it is only from one Autoload to another, etc.  As of yet I can't conceive of a work-around to this.

From the editor the reference will be:
godot windows tools 64_2019-02-28_19-15-31

In the release build.
test_2019-02-28_19-15-41

**Steps to reproduce:**

An Autoload doing this code seems to negate it's global reference in release builds.

```gdscript
func _ready():

	var arr = [1, 2, 3]

	for a in arr:
		a = 2
```

**Minimal reproduction project:**

Trimmed it down to this, make and run a release build of this project to recreate the issue.
[_Project.zip](https://github.com/godotengine/godot/files/2916194/_Project.zip)

### PR DESCRIPTION
**Pull request by bojidar-bg**
Also, spotted a potential bug related to `get_token_literal` usage, which I fixed.

Fixes #25334, fixes #26412, and potentially #25408.

### CODE REVIEW

**Comment by akien-mga** [comment]
Fixes #25408 and #25334?

**Comment by akien-mga** [comment]
Might also fix #26412 (poke @avencherus).

File: modules/gdscript/gdscript_parser.cpp [hunk 1]
```diff
@@ -7842,13 +7842,16 @@ void GDScriptParser::_check_block_types(BlockNode *p_block) {
...
-						if (!lh_type.has_type && check_types) {
-							if (op->arguments[0]->type == Node::TYPE_OPERATOR) {
-								_mark_line_as_unsafe(op->line);
+						if (check_types) {
+							if (!lh_type.has_type) {
+								if (op->arguments[0]->type == Node::TYPE_OPERATOR) {
+									_mark_line_as_unsafe(op->line);
+								}
+							}
+							if (lh_type.is_constant) {
```
Comment (akien-mga) [inline modules/gdscript/gdscript_parser.cpp]: What's surprising is that `lh_type` evaluates as a constant when it's a local variable though.

Comment (bojidar-bg) [reply modules/gdscript/gdscript_parser.cpp]: I don't think I managed to reproduce that one, but, maybe it fixed the others.

Comment (vnen) [reply modules/gdscript/gdscript_parser.cpp]: Not sure why it would be marked as constant. But the fix looks good to me.

**Comment by vnen** [comment]
Thanks!

**Comment by avencherus** [comment]
Nice save.  X)  Many thanks.

@akien-mga Yep, my issue ended up being a duplicate.  The release builds after this commit are not vulnerable to crashing with operations on the iterators anymore.  The core stuff in my project is at least running smoothly.

File: modules/gdscript/gdscript_parser.cpp (hunk 1 above)
Comment (akien-mga) [reply modules/gdscript/gdscript_parser.cpp]: I guess that was due to b24cb922?

**Pull request merged**